MetaMask/metamask-mobile
# chore(devDeps): bump metro and react-native

## **Description**

Update react-native and metro to receive patches.
- Update metro from `0.73` to `0.76` 
- Migrate from `metro-config` to `@react-native/metro-config`
- Update `react-native` from `0.71.15` to `0.72.14`
  - Port patch
  - Remove hermes patch
- Bump and unpin `@react-native-community/` packages
- Update `@react-native-community/netinfo` from `6.0.0` to `^9.5.0`
  - [Changelog](https://github.com/react-native-netinfo/react-native-netinfo/blob/master/CHANGELOG.md)
  - [Release notes](https://github.com/react-native-netinfo/react-native-netinfo/releases)
  - v10 breaking change:
    > netinfo now requires AbortController
- Update `@react-native-masked-view/masked-view`
  - [Release notes](https://github.com/react-native-masked-view/masked-view/releases) 
- Upgrade to Gradle version 8.0.1 and Android SDK 33


React Native [Upgrade](https://github.com/orgs/MetaMask/projects/60/views/6?pane=issue&itemId=41033345)
[Android SDK Update](https://github.com/react-native-community/cli/issues/1856)

## **Related issues**


- Fixes: https://github.com/MetaMask/mobile-planning/issues/1262
- #7759

## **Manual testing steps**

Node environment must be `18.18.2`
Supports Java version `17.0.10`

1. Check out branch
2. Run `yarn setup`
3. Run `yarn watch`
4. Run `yarn start:ios` - wait for app to build on iOS simulator
5. Run `yarn start:android` - wait for app to build Android emulator


## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
